### PR TITLE
Preparation for sending ProxyMessages to BackingStore

### PR DESCRIPTION
**Pull request by AliceLetitia**

### CODE REVIEW

File: src/runtime/api-channel.ts [hunk 1]
```diff
@@ -600,6 +602,10 @@ export abstract class PECInnerPort extends APIPort {
   ProxyMessage(@Mapped handle: StorageProxy,
                @Direct message: ProxyMessage<CRDTTypeRecord>,
                @LocalMapped callback: Consumer<Promise<boolean>>): void  {}
+  BackingProxyMessage(@Mapped handle: StorageProxy,
+                 @Direct message: ProxyMessage<CRDTTypeRecord>,
+                 @Direct entityId: string,
+                 @LocalMapped callback: Consumer<Promise<boolean>>): void{}
```
Comment (csilvestrini) [inline src/runtime/api-channel.ts]: Indenting seems a bit off.

Also, I believe that indenting up to the paren has fallen out of favour. I'd suggest:

```
BackingProxyMessage(
    @Mapped handle: StorageProxy,
    @Direct message: ProxyMessage<CRDTTypeRecord>,
    @Direct entityId: string,
    @LocalMapped callback: Consumer<Promise<boolean>>): void {}
```

Additionally, you need a space after `void` (and you may as well delete the extra space after `void` in `ProxyMessage` above)

File: src/runtime/particle-execution-context.ts [hunk 2]
```diff
@@ -169,6 +169,11 @@ export class ParticleExecutionContext implements StorageCommunicationEndpointPro
           }
         }
         message.id = id;
+        if (entityId != null) {
+          return  new Promise((resolve) =>
```
Comment (csilvestrini) [inline src/runtime/particle-execution-context.ts]: Delete extra space after `return`

Comment (csilvestrini) [reply src/runtime/particle-execution-context.ts]: And no parens needed around `resolve`

Comment (csilvestrini) [reply src/runtime/particle-execution-context.ts]: (and you could update the one below too)

File: src/runtime/particle-execution-context.ts [hunk 3]
```diff
@@ -169,6 +169,11 @@ export class ParticleExecutionContext implements StorageCommunicationEndpointPro
           }
         }
         message.id = id;
+        if (entityId != null) {
```
Comment (csilvestrini) [inline src/runtime/particle-execution-context.ts]: Switching on entityId seems a little strange. Might be worth saying something about that in a comment on this `onProxyMessage` function.

File: src/runtime/particle-execution-host.ts [hunk 4]
```diff
@@ -193,6 +194,16 @@ class PECOuterPortImpl extends PECOuterPort {
     this.SimpleCallback(callback, res);
   }
 
+  async onBackingProxyMessage(store: BackingStore<CRDTTypeRecord>, message: ProxyMessage<CRDTTypeRecord>, entityId: string, callback: number) {
+    if (!(store instanceof BackingStore)) {
+      this.onReportExceptionInHost(new SystemException(new Error('expected new-style store but found old-style store hooked up to new stack'), 'onProxyMessage', ''));
```
Comment (csilvestrini) [inline src/runtime/particle-execution-host.ts]: Suggestion: something simpler, like: `Expected BackingStore for onBackingProxyMessage`

File: src/runtime/particle-execution-host.ts [hunk 5]
```diff
@@ -193,6 +194,16 @@ class PECOuterPortImpl extends PECOuterPort {
...
+      return;
+    }
+    const res = await store.onProxyMessage(message, entityId);
+    this.SimpleCallback(callback, res);
+
```
Comment (csilvestrini) [inline src/runtime/particle-execution-host.ts]: Delete blank line

File: src/runtime/particle-execution-context.ts (hunk 3 above)
Comment (AliceLetitia) [reply src/runtime/particle-execution-context.ts]: New comment above the if statement. What do you think?

Comment (csilvestrini) [reply src/runtime/particle-execution-context.ts]: Looks good, thanks!

**Pull request merged**